client/src/main/java/tds/session/Extern.java

```diff
@@ -0,0 +1,45 @@
+package tds.session;
+
+/**
+ * configuration properties for the session system
+ */
+public class Extern {
+    private String clientName;
+    private String environment;
+
+    public static class Builder {
```

Comment:
I like them being immutable, and I was thinking that was maybe why you did the builder.  I think we need to come up with some general rules for when we think it makes sense to use the builder pattern.  It jumped out at me because I didn't notice that it was used in Exam already.  It was the first time for me seeing it so I figured it made sense to have the discussion on when we should use it and when we shouldn't.

Let's talk it through and come to a conclusion.